packages/ui/src/components/editor/panels/Hierarchy/prefabList/index.tsx

```diff
@@ -0,0 +1,158 @@
+    </button>
+  )
+}
+const ScenePrefabListItem = ({
+  categoryTitle,
+  categoryItems,
+  isCollapsed
+}: {
+  categoryTitle: string
+  categoryItems: PrefabShelfItem[]
+  isCollapsed: boolean
+}) => {
+  const open = useHookstate(categoryTitle === 'Empty')
+  return (
+    <>
+      <button
+        onClick={() => open.set((prev) => !prev)}
+        className="flex w-full cursor-pointer items-center justify-between bg-theme-primary px-4 py-2 text-white"
+      >
+        <span>{categoryTitle}</span>
+        {isCollapsed || open.value ? <IoIosArrowUp /> : <IoIosArrowDown />}
+      </button>
+      <div className={`${isCollapsed || open.value ? '' : 'hidden'}`}>
+        <ul className="w-full bg-theme-primary">
+          {categoryItems.map((item) => (
+            <PrefabListItem item={item} />
+          ))}
+        </ul>
+      </div>
+    </>
+  )
+}
+
+const usePrefabShelfCategories = (search: string) => {
+  useHookstate(getMutableState(PrefabShelfState)).value
+
+  if (!search) {
+    return getState(PrefabShelfState)
+  }
+
+  const searchRegExp = new RegExp(search, 'gi')
+
+  return getState(PrefabShelfState).filter(({ name }) => !!name.match(searchRegExp)?.length)
+}
+
+export function PrefabList() {
+  const { t } = useTranslation()
+  const search = useHookstate({ local: '', query: '' })
+  const searchTimeout = useRef<ReturnType<typeof setTimeout> | null>(null)
+  const validItems = usePrefabShelfCategories(search.query.value)
+  const inputReference = useRef<HTMLInputElement>(null)
+
+  const shelves: Record<string, PrefabShelfItem[]> = {}
+  for (const item of validItems) {
+    shelves[item.category] ??= []
+    shelves[item.category].push(item)
+  }
+  const shelveslist: string[] = []
+  Object.entries(shelves).map(([category, items]) => {
+    shelveslist.push(category)
+  })
+
+  const onSearch = (text: string) => {
+    search.local.set(text)
+    if (searchTimeout.current) clearTimeout(searchTimeout.current)
+    searchTimeout.current = setTimeout(() => {
+      search.query.set(text)
+    }, 50)
+  }
+
+  useEffect(() => {
+    inputReference.current?.focus()
+  }, [])
+
+  return (
+    <>
+      <div className="h-auto w-full overflow-x-hidden overflow-y-scroll bg-theme-primary">
+        <div className="p-2">
+          <h2 className="text-center uppercase text-white">{t('editor:layout.assetGrid.prefab')}</h2>
+          <StringInput
+            placeholder={t('editor:layout.assetGrid.prefab-search')}
+            value={search.local.value}
+            onChange={(val) => onSearch(val)}
+            inputRef={inputReference}
+          />
+        </div>
+      </div>
+      <PrefabListItem item={{ name: 'Empty', url: 'empty', category: '', detail: 'Basic scene entity' }} />
+      {Object.entries(shelves).map(([category, items]) => (
+        <ScenePrefabListItem categoryTitle={category} categoryItems={items} isCollapsed={!!search.query.value} />
+      ))}
+    </>
+  )
+}
+
+export default PrefabList
```

Comment:
this new prefab list is almost similar to [elementList](https://github.com/EtherealEngine/etherealengine/blob/1c68aaa125ad23ba0288d35bee8541eb50174c08/packages/ui/src/components/editor/panels/Properties/elementList/index.tsx)

The only difference here is the action performed when a prefab item is clicked from the list and the name of the list being prefab list.

We should make the element list dynamic to reduce code duplication.